src/utils/account-with-lockup.js

```diff
@@ -81,10 +81,15 @@ export async function transferAllFromLockup(missingAmount) {
 
     const lockedBalance = new BN(await this.wrappedAccount.viewFunction(lockupAccountId, 'get_locked_amount'))
     if (lockedBalance.eq(new BN(0))) {
-        // TODO: CHECK STAKING POOL BALANCE FIRST. OTHERWISE WON'T BE ABLET TO UNSELECT
+        const stakingPoolBalance = await this.wrappedAccount.viewFunction(lockupAccountId, 'get_known_deposited_balance')
+        if (stakingPoolBalance) {
+            throw new WalletError('Staking pool balance detected.', 'lockup.transferAllWithStakingPoolBalance')
+        }
```

Comment:
@vgrichina I now check the staking pool balance before attempting to unselect the staking pool. If the lockup balance is zero, but there's a staking pool balance (staked or pending), I show the above error on the UI instructing the user to unstake and withdraw all funds from the pool in order to transfer the remaining lockup balance.

Reply:
@Patrick1904 It's best to call the pool contract directly vs calling lockup contract (which might have out of date info). See https://github.com/near/core-contracts/tree/master/staking-pool#interface